tests/test_client.py

```diff
@@ -0,0 +1,373 @@
+                    sample_period
+                )
+                self.assertEqual(
+                    result[RECEIVED_PARAMS_KEY][VirusTotalApiClient._PARAM_REPEAT],
+                    sample_repeat
+                )
+                self.assertEqual(
+                    result[RECEIVED_PARAMS_KEY][VirusTotalApiClient._PARAM_NOTIFY_URL],
+                    sample_notifyurl
+                )
+                self.assertTrue(
+                    result[RECEIVED_PARAMS_KEY][VirusTotalApiClient._PARAM_NOTIFY_CHANGES_ONLY]
+                )
+
+                self.assertDictEqual(result, SAMPLE_FILE_RESCAN)
+
+
+            dxl_client.disconnect()
+
+
+    def test_ipreport(self):
+        with self.create_client(max_retries=0) as dxl_client:
+            # Set up client, and register mock service
+            vt_client = VirusTotalApiClient(dxl_client)
+            dxl_client.connect()
+
+            with MockVtService(dxl_client):
+
+                result = vt_client.ip_report(SAMPLE_IP)
+
+                self.assertDictEqual(result, SAMPLE_IP_ADDRESS_REPORT)
+
+            dxl_client.disconnect()
+
+
+    def test_urlreport(self):
+        with self.create_client(max_retries=0) as dxl_client:
+            # Set up client, and register mock service
+            vt_client = VirusTotalApiClient(dxl_client)
+            dxl_client.connect()
+
+            with MockVtService(dxl_client):
+                result = vt_client.url_report(SAMPLE_URL)
+
+                self.assertDictEqual(result, SAMPLE_URL_REPORT)
+
+            dxl_client.disconnect()
+
+
+    def test_urlreport_params(self):
+        with self.create_client(max_retries=0) as dxl_client:
+            # Set up client, and register mock service
+            vt_client = VirusTotalApiClient(dxl_client)
+            dxl_client.connect()
+
+            with MockVtService(dxl_client):
+                result = vt_client.url_report(
+                    SAMPLE_URL,
+                    scan="Anything",
+                    all_info="Anything"
+                )
+
+                self.assertTrue(
+                    result[RECEIVED_PARAMS_KEY][VirusTotalApiClient._PARAM_SCAN]
+                )
+
+                self.assertTrue(
+                    result[RECEIVED_PARAMS_KEY][VirusTotalApiClient._PARAM_ALLINFO]
+                )
+
+                self.assertDictEqual(result, SAMPLE_URL_REPORT)
+
+            dxl_client.disconnect()
+
+
+    def test_urlscan(self):
+        with self.create_client(max_retries=0) as dxl_client:
+            # Set up client, and register mock service
+            vt_client = VirusTotalApiClient(dxl_client)
+            dxl_client.connect()
+
+            with MockVtService(dxl_client):
+
+                result = vt_client.url_scan(SAMPLE_URL)
+
+                self.assertDictEqual(result, SAMPLE_URL_SCAN)
+
+            dxl_client.disconnect()
+
+
+class TestAddParams(BaseClientTest):
```

Comment:
I think the tests that you added above which cover the different parameters for the client methods would already implicitly cover these private add param helper methods. Given that, I'm not sure that there's much additional value to having the tests in this class. I'm okay with keeping this if you think it's worthwhile, though. What do you think?